src/MudBlazor/Components/Menu/MudMenu.razor.cs

```diff
@@ -752,5 +778,345 @@ public void Dispose()
+                    }
+                }
+                else
+                {
+                    await CloseAllMenusAsync();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Runs after component rendering. If the menu is open and no item is focused,
+        /// it automatically focuses the first enabled item in the list.
+        /// </summary>
+        protected override async Task OnAfterRenderAsync(bool firstRender)
+        {
+            await base.OnAfterRenderAsync(firstRender);
+
+            if (_openState.Value && _focusedIndex == -1)
+            {
+                await Task.Yield();
+                if (_menuItems.Count > 0)
+                {
+                    _focusedIndex = 0;
+                    await FocusItemAsync(_focusedIndex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Registers a new menu item or submenu with the current menu.
+        /// Ensures the item is only added once
+        /// </summary>
+        internal void RegisterItem(object item)
+        {
+            if (!_menuItems.Contains(item))
+            {
+                _menuItems.Add(item);
+            }
+        }
+
+        /// <summary>
+        /// Sets focus to the menu item at the specified index, if the index is valid.
+        /// </summary>
+        internal async Task FocusItemAsync(int index)
+        {
+            if (index >= 0 && index < _menuItems.Count)
+            {
+                var item = _menuItems[index];
+
+                // Retrieves the cref ElementRef associated with a menu item or submenu to allow focus control.
+                ElementReference elementRef = item switch
+                {
+                    MudMenuItem menuItem => menuItem.ElementReference,
+                    MudMenu menu => menu._menuItemActivator?.ElementReference ?? default,
+                    _ => default
+                };
+
+                if (elementRef.Context is not null)
+                {
+                    await elementRef.FocusAsync();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Subscribes to keyboard events for this menu using the <see cref="KeyInterceptorService"/>,
+        /// preventing default browser scrolling behaviour for certain keys.
+        /// </summary>
+        private async Task SubscribeToMenuKeyInterceptorAsync()
+        {
+            // Subscribe key interceptor to prevent default scrolling
+            var options = new KeyInterceptorOptions(
+                    "mud-list",
+                    [
+                        // prevent scrolling page
+                        new("ArrowDown", preventDown: "key+none"),
+                        new("ArrowUp", preventDown: "key+none"),
+                    ]);
+
+            await KeyInterceptorService.SubscribeAsync(_elementId, options, keyDown: HandleKeyDownAsync);
+        }
```

Comment:
This can return the task.

```suggestion
        private Task SubscribeToMenuKeyInterceptorAsync()
        {
            // Subscribe key interceptor to prevent default scrolling
            var options = new KeyInterceptorOptions(
                "mud-list",
                [
                    // prevent scrolling page
                    new("ArrowDown", preventDown: "key+none"),
                    new("ArrowUp", preventDown: "key+none"),
                ]);

            return KeyInterceptorService.SubscribeAsync(_elementId, options, keyDown: HandleKeyDownAsync);
        }
```